File: player.py
class Player:
    def __init__(self, name, objective):
        self.name = name              # Name of the player
        self.objective = objective    # Player's objective (reach my job)
        self.position = [0, 0]        # Player's position on the map (x, y)

    def move_left(self):
        """
        Move the player to the left (decreases the x-coordinate).
        """
        self.position[0] -= 1
        print(f"{self.name} moves left. New position: {self.position}")
    
    def move_right(self):
        """
        Move the player to the right (increases the x-coordinate).
        """
        self.position[0] += 1
        print(f"{self.name} moves right. New position: {self.position}")
    
    def move_forward(self):
        """
        Move the player forward (increases the y-coordinate).
        """
        self.position[1] += 1
        print(f"{self.name} moves forward. New position: {self.position}")
    
    def move_backward(self):
        """
        Move the player backward (decreases the y-coordinate).
        """
        self.position[1] -= 1
        print(f"{self.name} moves backward. New position: {self.position}")
    
    def jump(self):
        """
        Make the player jump (this could be more complex depending on game logic).
        """
        print(f"{self.name} jumps!")
    
    def dash(self):
        """
        Make the player dash, moving quickly forward by 2 units.
        """
        self.position[1] += 2
        print(f"{self.name} dashes forward. New position: {self.position}")

# Example usage in main.py would be similar to:
# player.move_left(), player.jump(), etc.

File: player.cs
using System;

public enum PlayerStatus
{
    Normal,
    Stopped,
    SlowMotion,
    Invisible,
    SpeedBoost,
    Blocked
}

public class Player
{
    public string Name { get; set; }
    public string Objective { get; set; }
    public int[] Position { get; set; }
    public PlayerStatus Status { get; set; }

    public Player(string name, string objective)
    {
        Name = name;
        Objective = objective;
        Position = new int[] { 0, 0 };  // x, y coordinates
        Status = PlayerStatus.Normal;
    }

    public void MoveLeft()
    {
        if (IsBlockedOrStopped())
            return;

        Position[0] -= 1;
        Console.WriteLine($"{Name} moves left. New position: [{Position[0]}, {Position[1]}]");
    }

    public void MoveRight()
    {
        if (IsBlockedOrStopped())
            return;

        Position[0] += 1;
        Console.WriteLine($"{Name} moves right. New position: [{Position[0]}, {Position[1]}]");
    }

    public void MoveForward()
    {
        if (IsBlockedOrStopped())
            return;

        Position[1] += 1;
        Console.WriteLine($"{Name} moves forward. New position: [{Position[0]}, {Position[1]}]");
    }

    public void MoveBackward()
    {
        if (IsBlockedOrStopped())
            return;

        Position[1] -= 1;
        Console.WriteLine($"{Name} moves backward. New position: [{Position[0]}, {Position[1]}]");
    }

    public void Jump()
    {
        if (IsBlockedOrStopped())
            return;

        Console.WriteLine($"{Name} jumps!");
    }

    public void Dash()
    {
        if (IsBlockedOrStopped())
            return;

        Position[1] += 2;
        Console.WriteLine($"{Name} dashes forward. New position: [{Position[0]}, {Position[1]}]");
    }

    public void ChangeStatus(PlayerStatus newStatus)
    {
        Status = newStatus;
        Console.WriteLine($"{Name} is now {Status}.");
    }

    private bool IsBlockedOrStopped()
    {
        if (Status == PlayerStatus.Blocked)
        {
            Console.WriteLine($"{Name} is blocked and cannot move.");
            return true;
        }
        if (Status == PlayerStatus.Stopped)
        {
            Console.WriteLine($"{Name} is stopped and cannot move.");
            return true;
        }
        return false;
    }
}

// Example usage
public class Program
{
    public static void Main(string[] args)
    {
        Player player = new Player("Alex", "Find a job");

        player.MoveForward(); // Normal state
        player.ChangeStatus(PlayerStatus.Blocked);
        player.MoveForward(); // Blocked state
        player.ChangeStatus(PlayerStatus.SpeedBoost);
        player.Dash(); // Speed boost state
    }
}
